src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,118 @@
+package mate.academy.springboot.swagger.controller;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import mate.academy.springboot.swagger.dto.request.ProductRequestDto;
+import mate.academy.springboot.swagger.dto.response.ProductResponseDto;
+import mate.academy.springboot.swagger.mapper.RequestDtoMapper;
+import mate.academy.springboot.swagger.mapper.ResponseDtoMapper;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.ProductService;
+import mate.academy.springboot.swagger.service.util.SortRequestParser;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final RequestDtoMapper<ProductRequestDto, Product> requestMapper;
+    private final ResponseDtoMapper<Product, ProductResponseDto> responseMapper;
+
+    public ProductController(ProductService productService,
+                             RequestDtoMapper<ProductRequestDto, Product> requestMapper,
+                             ResponseDtoMapper<Product, ProductResponseDto> responseMapper) {
+        this.productService = productService;
+        this.requestMapper = requestMapper;
+        this.responseMapper = responseMapper;
+    }
+
+    @PostMapping("/create")
+    @ApiOperation("Endpoint to create new products")
+    public ProductResponseDto create(@RequestBody @Valid ProductRequestDto productRequestDto) {
+        return responseMapper.mapToDto(productService.create(
+                requestMapper.mapToModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation("Endpoint to get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return responseMapper.mapToDto(productService.get(id)
+                .orElseThrow(() -> new NoSuchElementException("Can`t find product with id:"
+                        + id)));
```

Comment:
Let's move throwing an exception inside service method. 
Ideally the controller layer shouldn't have any business logic.